app/models/record.rb

```diff
@@ -1,6 +1,61 @@
+# frozen_string_literal: true
+
 class Record < ApplicationRecord
   audited only: :updated_at
+  before_create :convert_maz_json_to_sva_json
+
+  private
+
+    def convert_maz_json_to_sva_json
+      news_data = parse_single_news_from_json(json_data)
+      self.sva_json_data = { news: [news_data] }
+    end
+
+    def parse_single_news_from_json(json_hash)
+      result = {
+        external_id: json_hash.dig("_id"),
+        title: json_hash.dig("title", "value"),
+        publication_date: json_hash.dig("publication_date", "value"),
+        published_at: json_hash.dig("publish_date", "value"),
+        show_publish_date: json_hash.dig("show_publish_date", "value"),
+        news_type: "news article",
+        address: parse_address(json_hash),
+        content_blocks: ContentBlockParser.perform(json_hash)
+      }
+      result.merge(parse_url(json_data))
+
+    end
+
+    def parse_address(json_hash)
+      if json_hash.dig("geo_location", "value", "latitude").blank? || json_hash.dig("geo_location", "value", "longitude").blank?
+        {
+          city: json_hash.dig("location_name", "value")
+        }
+      else
+        {
+          city: json_hash.dig("location_name", "value"),
+          geo_location: {
```

Comment:
es wird in Zeile 30 geprüft, ob latitude oder longitude blank sind. es kann also sein, dass hier einer von beiden immer noch blank wäre. dann ist es auch keine valide geo-location.
vllt. besser oben mit `&&` statt `||` prüfen?

`location_name` könnte auch blank sein, oder? darauf wird hier gar nicht geprüft.